relayer/relays/parachain/types.go

```diff
@@ -1,45 +1,131 @@
 package parachain
 
 import (
+	"fmt"
+	"math/bits"
+
 	"github.com/snowfork/go-substrate-rpc-client/v4/types"
 	"github.com/snowfork/snowbridge/relayer/chain/relaychain"
 	"github.com/snowfork/snowbridge/relayer/crypto/merkle"
 )
 
 // A Task is a bundle of items needed to submit commitments to Ethereum
 type Task struct {
-	ParaID                  uint32
-	BlockNumber             uint64
-	Header                  *types.Header
-	Commitments             map[ChannelID][]Commitment
-	BasicChannelBundleProof *MerkleProof
-	ProofInput              *ProofInput
-	ProofOutput             *ProofOutput
+	ParaID                        uint32
+	BlockNumber                   uint64
+	Header                        *types.Header
+	BasicChannelProofs            *[]MerkleProof
+	IncentivizedChannelCommitment *IncentivizedChannelCommitment
+	ProofInput                    *ProofInput
+	ProofOutput                   *ProofOutput
 }
 
 // A Commitment is data provably attested to by polkadot. The commitment hash
 // is contained in a parachain header. Polkadot validator nodes attest that the header
 // is genuine.
-type Commitment struct {
+type IncentivizedChannelCommitment struct {
 	Hash types.H256
-	Data interface{}
+	Data IncentivizedOutboundChannelMessageBundle
 }
 
-func NewCommitment(hash types.H256, data interface{}) Commitment {
-	return Commitment{
+func NewIncentivizedChannelCommitment(hash types.H256, data IncentivizedOutboundChannelMessageBundle) IncentivizedChannelCommitment {
+	return IncentivizedChannelCommitment{
```

Comment:
As noted in comments for the `beefy-listener`, you can probably just return `&IncentivizedChannelCommitment` here.